### PR TITLE
remove notion of ProcedureID

### PR DESCRIPTION
**Pull request by jlapacik**

### CODE REVIEW

File: functions/transformations/join.go [hunk 1]
```diff
@@ -211,13 +212,6 @@ func (s *MergeJoinProcedureSpec) Copy() plan.ProcedureSpec {
 	return ns
 }
 
-func (s *MergeJoinProcedureSpec) ParentChanged(old, new plan.ProcedureID) {
```
[the remark this answers is not in the document]
Comment (jlapacik) [reply functions/transformations/join.go]: It was dead code on this branch, but you're right it was used by the old planner.

[the remark this answers is not in the document]
Comment (jlapacik) [reply functions/transformations/join.go]: Yeah that's what I would like to do. Currently it needs the mapping of table name and corresponding dataset ID.

[the remark this answers is not in the document]
Comment (jlapacik) [reply functions/transformations/join.go]: What if `plan.Administration` had a map of `ProcedureSpecs` to `PlanNodes`? Then we could have the table names of join point to their corresponding plan nodes.

[the remark this answers is not in the document]
Comment (jlapacik) [reply functions/transformations/join.go]: The planner then has to update an Administration during planning.

[the remark this answers is not in the document]
Comment (jlapacik) [reply functions/transformations/join.go]: The other option is to bring back the `ParentAwareProcedureSpec` type.

**Comment by wolffcm** [comment]
Looks great.  Always nice to see more lines of code deleted than were added.

**Pull request merged**